engines/extpoints/interfaces.go

```diff
@@ -0,0 +1,10 @@
+// Package extpoints provides methods that engine plugins can register their
+// implements with as an import side-effect.
+package extpoints
+
+import (
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+)
+
+type EngineProvider func(environment *runtime.Environment, options *engines.EngineOptions) (engines.Engine, error)
```

Comment:
Does this work? Ie. I think the go-extpoints generator thing strictly looks for interfaces, but I could be wrong?

Ahh, seems I was wrong... This should work:
https://github.com/progrium/go-extpoints/blob/master/main.go#L35